### PR TITLE
Allow optional braces after old style while and for

### PR DESCRIPTION
**Pull request by tgodzik**
Fixes https://github.com/scalameta/scalafmt/issues/2790

### CODE REVIEW

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala [hunk 1]
```diff
@@ -2133,19 +2133,21 @@ class FormatOps(
               def splits = Some(getSplitsForIf(ft, nft, t))
               def rightBrace = blockLast(t.thenp)
             })
-          case t: Term.For if !nft.right.is[T.KwDo] =>
-            // unsupported except for right brace
+          case Term.For(_, t: Term.Block)
```
Comment (kitbellew) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala:2136]: looks like we now have extra conditions. is that equivalent? keep in mind that right brace method may have been used not just for formatting but for rewrite, maybe end marker, and possibly something else. can we do a quick search on usage of `rightBrace` and see if there are cases we'd forgotten to cover with tests?

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala [hunk 2]
```diff
@@ -2133,19 +2133,21 @@ class FormatOps(
...
+              if !nft.right.is[T.KwDo] && isBlockStart(t, nft) =>
             Some(new OptionalBracesRegion {
-              def owner = None
-              def splits = None
```
[the remark this answers is not in the document]
Comment (kitbellew) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala:2140]: in #2420, the last comment talks about following up with "for" and "while" after fixing in the parser, perhaps we simply never did.

File: scalafmt-tests/src/test/resources/scala3/OptionalBraces.stat [hunk 3]
```diff
@@ -2208,3 +2208,35 @@ def foo2: Int =
       44
    end if
 end foo2
+<<< 2790 for loops indentation
+def update(tpf: Float): Unit =
+  val s = stateModifier(currentState, tpf)
+  currentState = s.state
+  processCommands(s.commands)
+  for (scene <- scenes.all)
```
Comment (kitbellew) [inline scalafmt-tests/src/test/resources/scala3/OptionalBraces.stat:2216]: can you add another loop with a single statement in the body? it looks like plain old scala2 code and should not use significant indentation. same with while.

File: scalafmt-tests/src/test/resources/scala3/OptionalBraces.stat [hunk 4]
```diff
@@ -2208,3 +2208,35 @@ def foo2: Int =
       44
    end if
 end foo2
+<<< 2790 for loops indentation
+def update(tpf: Float): Unit =
+  val s = stateModifier(currentState, tpf)
```
Comment (kitbellew) [inline scalafmt-tests/src/test/resources/scala3/OptionalBraces.stat:2213]: optional: i don't think we need these three lines. just the loops is probably enough.

**Review by kitbellew — CHANGES_REQUESTED** [review]

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (tgodzik) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala:2140]: Och, that might be right :sweat_smile:

**Comment by kitbellew** [comment]
Submitted in #2803.

**Pull request closed without merging**